src/main/java/com/google/idea/perf/TracerView.kt

```diff
@@ -83,12 +86,25 @@ class TracerView(parentDisposable: Disposable) : JBPanel<TracerView>() {
         layout = BoxLayout(this, BoxLayout.Y_AXIS)
 
         // Command line.
-        val commandLine = JBTextField().apply {
+        commandLine = JBTextField().apply {
             maximumSize = Dimension(Integer.MAX_VALUE, minimumSize.height)
             addActionListener { e ->
                 text = ""
                 controller.handleRawCommandFromEdt(e.actionCommand)
             }
+            document.addDocumentListener(object: DocumentListener {
+                override fun changedUpdate(e: DocumentEvent?) {
```

Comment:
When overriding a Java method, Kotlin will by default use the nullable type `DocumentEvent?` because it can't prove that the Java code never passes null.

However, in this case we happen to know that `e` will always be non-null, so we can just remove the `?` from the type (and then remove the `?: 0` snippets below).